include/pybind11/cast.h

```diff
@@ -960,9 +960,15 @@ template <typename type> class type_caster<std::reference_wrapper<type>> {
 private:
     using caster_t = make_caster<type>;
     caster_t subcaster;
-    using subcaster_cast_op_type = typename caster_t::template cast_op_type<type>;
-    static_assert(std::is_same<typename std::remove_const<type>::type &, subcaster_cast_op_type>::value,
-            "std::reference_wrapper<T> caster requires T to have a caster with an `T &` operator");
+    using reference_t = typename std::add_lvalue_reference<type>::type;
+    using subcaster_cast_op_type =
+        typename caster_t::template cast_op_type<reference_t>;
+
+    static_assert((std::is_same<typename std::remove_const<type>::type &,
+                             subcaster_cast_op_type>::value ||
+                std::is_same<reference_t, subcaster_cast_op_type>::value),
+               "std::reference_wrapper<T> caster requires T to have a caster "
+               "with an operator `T &` or `const T&`");
```

Comment:
Since this line isn't as long as this one below:

```
static handle cast(const std::reference_wrapper<type> &src, return_value_policy policy, handle parent) {
```

It's probably more clear if we put it like this? (also with correct indentation and with `const T &` instead of `const T&`, like the rest of pybind11's codebase)

```suggestion
    static_assert(std::is_same<typename std::remove_const<type>::type &, subcaster_cast_op_type>::value ||
                  std::is_same<reference_t, subcaster_cast_op_type>::value,
                  "std::reference_wrapper<T> caster requires T to have a caster with an "
                  "`operator T &()` or `operator const T &()`");
```

Reply:
Done.